model/log/service_v0.go

```diff
@@ -115,28 +124,35 @@ func (s *logServiceV0) getLogChunks(ctx context.Context, logNames []string) (map
 		if err != nil {
 			return nil, 0, 0, errors.Wrapf(err, "parsing chunk key '%s'", chunkKey)
 		}
-		logChunks[logName] = append(logChunks[logName], chunk)
+
+		chunks, ok := logChunksByName[logName]
```

Comment:
is the idea here that the result of the bucket.List is always sorted and we just need to make sure to preserve that order?
If so maybe an alterative strategy could be to append the log names to a slice as we're iterating the List and putting the chunks into a logChunks map. Then to iterate the map with the keys from the slice, like
```go
var allChunks []chunkInfo
for _, logName := range logNames {
    allChunks = append(allChunks, logChunks[logName]...)
}
```
This would have an advantage that we wouldn't need to do the pointer stuff (which took me some time to follow)

Reply:
Great idea, I am all for simplifying the logic!